src/mysql_protocol.erl

```diff
@@ -1220,6 +1214,86 @@ recv_packet(SockModule, Socket, Timeout, ExpectSeqNum, Acc) ->
             {error, Reason}
     end.
 
+-spec send_file(module(), term(), Filename :: binary(), AllowedPaths :: [binary()],
+                SeqNum :: integer()) ->
+    {ok | {error, term()}, NextSeqNum :: integer()}.
```

Comment:
Better be explicit and say `{error, not_allowed}` in the spec here.

Reply:
`not_allowed` is not the only error that could be returned from this function, but also any error that is returned from `file:open/2`.